### ISSUE TITLE
app_files ordering

### ISSUE THREAD

**Issue by DougPuchalski** [clayallsopp/motion-require#9]
It appears that files are added at the very end of the app_files array. Do you have any thoughts as to how to control ordering? I'm not sure it's always possible to add a motion_require statement that will take care of the dependencies. 

In any event, I'm wondering if the sugarcube-style insert_point to always add right before the app would be workable?

**Comment by clayallsopp** [comment]
Hmm, is the `app.files` ordering significant? As in, I was under the (maybe wrong) assumption that if you setup `app.files_dependencies` correctly, then RubyMotion will evaluate those requirements first and then just append everything else from `app.files` afterwards in the order of `app.files`. Or maybe do you have an example of how just appending would break the build?

And I'm not totally sure I understand what you mean by "add right before the app"?

Thanks for bringing it up though, I use it within my internal projects but haven't exhaustively tested it :)

**Comment by DougPuchalski** [comment]
The order is definitely important, if you don't have all your dependencies specified. The commonly used pattern is here:

https://github.com/rubymotion/sugarcube/blob/master/lib/sugarcube.rb

A little old-school, but it makes sure your app stays at the end of the app_files, and removed the need to declare any dependencies from the app. Without it, there's a lot of wasted work adding dependencies manually.

So basically, if rather than putting them on the end, you insert them before the first "./app/" line we should probably be good.

**Comment by kastiglione** [comment]
Just added a PR for this, #10.
